[ADD] inventory: add res at confirmation doc (#7876)
odoo/documentation: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation.rst

===========================
At confirmation reservation
===========================

The *At Confirmation* reservation method reserves products **only** when a sales order (SO) is
confirmed, **and** if enough stock of the products included in the :abbr:`SO (sales order)` is

> already available.

Reviewer: Can use an abbreviation declaration at the beginning of the doc after the h1, for SO and others:
```
.. |SO| replace:: :abbr:`SO (Sales Order)`
```

> already available.

Author: I like to write it out upon first mention in the doc, like sales order (SO), and then abbreviate after the first mention. However, if that's not how I should be doing it let me know. I did add in the |SO| abbreviation declaration at the top though.

.. seealso::
   :doc:`About reservation methods <../reservation_methods>`

Configuration
=============

To set the reservation method to *At Confirmation*, navigate to :menuselection:`Inventory app -->

> Configuration --> Operations Types`.

Author: Suggestion implemented ✅

In the :guilabel:`General` tab, locate the :guilabel:`Reservation Method` field, and select

> :guilabel:`At Confirmation`.

Author: Suggestion implemented ✅

.. image:: at_confirmation/at-confirmation-operations-type.png
   :align: center
   :alt: Reservation method field on delivery order operation type form.

Workflow
========

To see the *At Confirmation* reservation method in action, create a new :abbr:`SO (sales order)` by
navigating to :menuselection:`Sales app --> New`.

Add a customer in the :guilabel:`Customer` field, then, in the :guilabel:`Order Lines` tab, click
:guilabel:`Add a product`, and select a product to add to the quotation from the drop-down menu.

> Finally, in the :guilabel:`Quantity` column, adjust the desired quantity of the product to sell.

Author: Suggestion implemented ✅

Once ready, click :guilabel:`Confirm` to confirm the sales order.

> Click the green :guilabel:`forecast graph` icon on the product line to reveal the product's

Reviewer 2: ```suggestion
Click the green :guilabel:` 📈 (forecast graph)` icon on the product line to reveal the product's
```
You can select a different emoji if you think there's a better one.

Author: That one works for me! Suggestion implemented ✅

:guilabel:`Availability` tooltip. This tooltip reveals the reserved number of units for this order,
which, in this case, is `100 Units`.

.. image:: at_confirmation/at-confirmation-availability-tooltip.png
   :align: center
   :alt: Confirmed sales order with product availability tooltip selected.

.. admonition:: Forecasted Report

   To see all the factors that affect product reservation, click the :guilabel:`View Forecast`

>    internal link arrow to view the :guilabel:`Forecasted Report` page.

Author: Let me look into that one -- as far as I understood, I could always see the link despite availability. I'll do some testing and update accordingly.

>    internal link arrow to view the :guilabel:`Forecasted Report` page.

Author: So, I did some testing and found I am able to see the link with 0 product in stock, and even negative forecasted stock. However, I did find that the availability tooltip changes based on the quantity on hand. So I added this admonition below:

.. note::
   If there is **not** sufficient quantity of stock for the product included in the |SO|, the
   :guilabel:`📈 (forecast graph)` icon is red, instead of green.

   Instead of revealing the reserved number of units for the order, the :guilabel:`Availability`
   tooltip reads :guilabel:`Available`, and reveals the available number of units (e.g., `0 Units`).

Thanks for pointing this out! Suggestion implemented ✅

   The :guilabel:`Forecasted Report` displays forecast information about the product(s) included in
   the sales order; namely, any live receipts of the product, and any active sales orders, listed in
   the :guilabel:`Used By` column. See how each order will be fulfilled in the

>    :guilabel:`Replenishment` column.

Author: Suggestion implemented ✅

   Additionally, the :guilabel:`Forecasted` quantity is calculated at the top of the page, by adding

>    the **On Hand** and **Incoming** quantity, and subtracting the **Outgoing** quantity, as shown
>    below:

Reviewer 3: ```suggestion
   the :guilabel:`On Hand` and :guilabel:`Incoming` quantity, and subtracting the
   :guilabel:`Outgoing` quantity, as shown below:
```
suggesting the use of GUIs here because they appear on the screen/UI. Same reasoning behind your usage of GUIs for 'Forecasted'

Author: Good catch. Suggestion implemented ✅

   .. image:: at_confirmation/at-confirmation-forecasted-equation.png
      :align: center
      :alt: Forecasted quantity equation from the Forecasted Report page.

   If one order should be prioritized over another order, click the :guilabel:`Unreserve` button on
   the according order line in the :guilabel:`Replenishment` column.

To deliver the products, click the :guilabel:`Delivery` smart button at the top of the sales order
form. To confirm that the reservation worked properly, ensure that the :guilabel:`Product

> Availability` field reads `Available` (in green text), and the number in the :guilabel:`Demand` and

Reviewer 2: ```suggestion
Availability` field reads `Available` (in green text), and the numbers in the :guilabel:`Demand` and
```

Author: Suggestion implemented ✅

===========================
At confirmation reservation
===========================

.. _inventory/reservation_methods/at-confirmation:

.. |SO| replace:: :abbr:`SO (Sales Order)`

> The *At Confirmation* reservation method reserves products **only** when a sales order (SO) is

Reviewer: Not a proper noun.
```suggestion
The *at confirmation reservation method* reserves products **only** when a sales order (SO) is
```

Author: That's how it appears in the UI, but I will take this suggestion and apply it throughout the doc. Suggestion implemented ✅

confirmed, **and** if enough stock of the products included in the |SO| is already available.

> .. seealso::
>    :doc:`About reservation methods <../reservation_methods>`
> 

Reviewer 4: Suggestion to move this seealso to the bottom of the doc :)
```suggestion
.. seealso::
   :doc:`About reservation methods <../reservation_methods>`

```

Author: Suggestion implemented ✅

To set the reservation method to *At Confirmation*, navigate to :menuselection:`Inventory app -->
Configuration --> Operations Types`. Then, select the desired :guilabel:`Operation Type` to
configure, or create a new one by clicking :guilabel:`New`.

In the :guilabel:`General` tab on the operation type form, locate the :guilabel:`Reservation Method`
field, and select :guilabel:`At Confirmation`.

.. image:: at_confirmation/at-confirmation-operations-type.png
   :align: center
   :alt: Reservation method field on delivery order operation type form.

Workflow
========

To see the *At Confirmation* reservation method in action, create a new |SO| by navigating to
:menuselection:`Sales app --> New`.

Add a customer in the :guilabel:`Customer` field. Then, in the :guilabel:`Order Lines` tab, click
:guilabel:`Add a product`, and select a product to add to the quotation from the drop-down menu.
Finally, in the :guilabel:`Quantity` column, adjust the desired quantity of the product to sell.

Once ready, click :guilabel:`Confirm` to confirm the sales order.

> Click the green :guilabel:`📈 (forecast graph)` icon on the product line to reveal the product's
> :guilabel:`Availability` tooltip. This tooltip reveals the reserved number of units for this order,
> which, in this case, is `100 Units`.

Reviewer: - icon name (see screenshot, revealed in dev tools)
- icon may or may not be green for the user, would remove this status indicator (what you did with the **Note** is ok too 👍 )
- remove reference to the screenshot; the writing should be able to stand on it's own *unless* we're writing in 
this way with an **example** admonition block
- add gui for **Reserved** since it is a line item label on the tooltip
![Screenshot 2024-03-14 at 1 23 43 PM](https://github.com/odoo/documentation/assets/5496809/a8b9ba16-3fdf-4eb8-b991-07a9e929a5b1)

```suggestion
Click the :guilabel:`📈 (area graph)` icon on the product line to reveal the product's
:guilabel:`Availability` tooltip, which reveals the :guilabel:`Reserved` number of units for this order.
```

Author: Suggestion implemented ✅

.. note::
   If there is **not** sufficient quantity of stock for the product included in the |SO|, the

>    :guilabel:`📈 (forecast graph)` icon is red, instead of green.

Reviewer: ```suggestion
   :guilabel:`📈 (area graph)` icon is red, instead of green.
```

Author: Suggestion implemented ✅

   Instead of revealing the reserved number of units for the order, the :guilabel:`Availability`
   tooltip reads :guilabel:`Available`, and reveals the available number of units (e.g., `0 Units`).

.. image:: at_confirmation/at-confirmation-availability-tooltip.png
   :align: center
   :alt: Confirmed sales order with product availability tooltip selected.

.. admonition:: Forecasted Report

   To see all the factors that affect product reservation, click the :guilabel:`View Forecast`

>    internal link arrow to view the :guilabel:`Forecasted Report` page.

Reviewer: nitpick 😉 
```suggestion
   internal link arrow to view the :guilabel:`Forecasted Report` dashboard.
```

Author: Suggestion implemented ✅
